### PR TITLE
add q parameter for free-text search

### LINKED ISSUE #293: Support for the STAC API free-text extension
The [STAC API free-text extension](https://github.com/stac-api-extensions/freetext-search?tab=readme-ov-file) enables a user to provide a parameter `q` with some free-text search query parameters. This is perhaps redundant with the filter extension, but it is much easier to write a text query using the `q` parameter. e.g. `landsat OR sentinel` to get records that contain the term `sentinel` or `landsat`, `landsat AND sentinel` to get records that contain both terms.

Adding support for the `q` parameter would be particularly useful for improving the collection search functionality!

### PR DESCRIPTION
**Pull request by hrodmn**
This adds the `q` parameter for free-text search. It will be pretty slow for item-level search because it has to compute the `tsvectors` on-the-fly, but it works great for collection-level search where the number of records should be limited.

Resolves #293 (for collection-level search at least)

### CODE REVIEW

File: src/pypgstac/pyproject.toml [hunk 1]
```diff
@@ -46,7 +46,7 @@ dev = [
     "black>=21.7b0",
     "mypy>=0.910",
     "types-orjson==0.1.1",
-    "types-pkg-resources",
+    "types-setuptools",
```
Comment (hrodmn) [inline src/pypgstac/pyproject.toml:49]: I got an error when I built the docker containers saying `types-pkg-resources` had been "pulled" and to replace it with `types-setuptools`

File: src/pgstac/sql/004_search.sql [hunk 2]
```diff
@@ -127,6 +127,32 @@ CREATE OR REPLACE FUNCTION partition_query_view(
...
+    RETURNS tsquery
+    AS $$
+DECLARE
+    processed_text text;
+BEGIN
+    -- replace commas outside quoted text with " | "
+    processed_text := regexp_replace(input, ',(?=(?:[^"]*"[^"]*")*[^"]*$)', ' | ', 'g');
+
+    -- replace the logical operators with tsquery equivalents
+    processed_text := regexp_replace(processed_text, '\s+AND\s+', ' & ', 'g');
```
Comment (bitner) [inline src/pgstac/sql/004_search.sql:140]: can we make this case insensitive?

File: CHANGELOG.md
Comment (hrodmn) [inline CHANGELOG.md:1]: I think my editor automatically applied some formatting changes to this file. I can roll them back if you want.

File: src/pgstac/sql/004_search.sql [hunk 3]
```diff
@@ -139,8 +165,10 @@ DECLARE
     sdate timestamptz;
     edate timestamptz;
     filterlang text;
-    filter jsonb := j->'filter';
+    filter jsonb = j->'filter';
```
Comment (bitner) [inline src/pgstac/sql/004_search.sql:168]: this should stay ":=" for assignment in plpgsql

**Comment by hrodmn** [comment]
@bitner I had to beef up `q_to_tsquery` a bit to avoid replacing terms inside of quoted statements. It's all case-insensitive now and passing a more robust set of tests!

**Review by bitner — APPROVED** [review]

**Pull request merged**